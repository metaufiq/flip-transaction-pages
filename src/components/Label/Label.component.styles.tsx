import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    padding: 10,
    borderWidth: 1,
    borderRadius: 10
  },
  value:{
    fontWeight: 'bold'
  }
})

export default styles;