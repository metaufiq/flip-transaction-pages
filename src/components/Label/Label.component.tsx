import React from "react";
import { Text, TextStyle, View, ViewStyle } from "react-native";
import { colors } from "../../configs/themes";
import styles from "./Label.component.styles";

import { Props, Type } from "./Label.component.types";

const _getBorderColor = (type: Type):ViewStyle['borderColor']=>{
  switch (type) {
    case 'PRIMARY':
      return colors.primary
    case 'SUCCESS':
      return colors.success
    default:
      return colors.primary
  }
}

const _getBackgroundColor = (type: Type):ViewStyle['borderColor']=>{
  switch (type) {
    case 'PRIMARY':
      return 'white'
    case 'SUCCESS':
      return colors.success
    default:
      return 'white'
  }
}

const _getTextColor = (type: Type):TextStyle['color']=>{
  switch (type) {
    case 'PRIMARY':
      return 'black'
    case 'SUCCESS':
      return 'white'
    default:
      return 'black'
  }
}

const _getContainerStyle = (type: Type)=>(
  {
    borderColor: _getBorderColor(type),
    backgroundColor: _getBackgroundColor(type)
  }
)

const _getValueStyle = (type: Type)=>(
  {
    color: _getTextColor(type)
  }
)

const Label = ({type, value}: Props)=>{
  return(
    <View style={[styles.container, _getContainerStyle(type)]}>
      <Text style={[styles.value, _getValueStyle(type)]}>{value}</Text>
    </View>
  )
}

export default Label;