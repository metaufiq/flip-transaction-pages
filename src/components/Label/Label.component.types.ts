export type Type = 'PRIMARY' | 'SUCCESS'

export interface Props{
  value: string,
  type: Type
}