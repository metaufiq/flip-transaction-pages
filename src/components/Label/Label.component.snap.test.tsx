import { assertSnapshots } from "../../../test";
import Label from ".";
import { Props } from "./Label.component.types";

describe('LabelSnapshot', () => {

  const mockProps:Props = {
    type: 'PRIMARY',
    value: 'value'
  };

  const configs = [
    {
      props: mockProps,
      desc: 'should render primary label'
    },
    {
      props: {...mockProps, type: 'SUCCESS'},
      desc: 'should render success label'
    },
  ];

  assertSnapshots(Label, configs);
});
