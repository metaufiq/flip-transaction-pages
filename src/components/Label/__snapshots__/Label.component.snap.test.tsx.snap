// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LabelSnapshot should render primary label 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "borderWidth": 1,
        "padding": 10,
      },
      Object {
        "backgroundColor": "white",
        "borderColor": "#fd6542",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "bold",
        },
        Object {
          "color": "black",
        },
      ]
    }
  >
    value
  </Text>
</View>
`;

exports[`LabelSnapshot should render success label 1`] = `
<View
  style={
    Array [
      Object {
        "borderRadius": 10,
        "borderWidth": 1,
        "padding": 10,
      },
      Object {
        "backgroundColor": "#4fb785",
        "borderColor": "#4fb785",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "fontWeight": "bold",
        },
        Object {
          "color": "white",
        },
      ]
    }
  >
    value
  </Text>
</View>
`;
